charts/kyuubi/templates/kyuubi-podmonitor.yaml

```diff
@@ -0,0 +1,31 @@
+{{/*
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/}}
+
+{{- if and .Values.server.prometheus.enabled (eq .Values.metricsReporters "PROMETHEUS") .Values.podMonitor.enabled }}
```

Comment:
How about we add some api check, like `if .Capabilities.APIVersions.Has "monitoring.coreos.com/v1/PodMonitor"`. You can find this is helm doc.

https://helm.sh/docs/chart_template_guide/builtin_objects/#:~:text=Capabilities.APIVersions%20is%20a%20set%20of%20versions.%20Capabilities.APIVersions.Has%20%24version,resource%20%28e.g.%2C%20apps%2Fv1%2FDeployment%29%20is%20available%20on%20the%20cluster.

> Capabilities.APIVersions.Has $version indicates whether a version (e.g., batch/v1) or resource (e.g., apps/v1/Deployment) is available on the cluster.

Reply:
Added monitoring API check.

Reply:
Am I right that Prometheus Operator and related CRDs are required to be installed for `PrometheusRule`, `PodMonitor` and `ServiceMonitor` to be used?
If so, it makes sense and is enough to just disable resource creation by default instead of making API check, i.e. set the following default values and revert API checks:
```yaml
podMonitor:
  enabled: false
  ...

serviceMonitor:
  enabled: false
  ...

prometheusRule:
  enabled: false
  ...
```
It is more preferable behavior to create the resource if `enabled=true`, even if the resource cannot be used in the Kubernetes cluster. For example, a user can deploy Kyuubi helm chart with enabled `ServiceMonitor` (ServiceMonitor resource will be created) to the cluster without Prometheus Operator, add Prometheus Operator later and everything will work as expected. In the current implementation with API checks a user will need to run `helm upgrade` command explicitly to trigger ServiceMonitor resource creation.

Reply:
Yeah, I believe that disabling resource creation by default instead of creating API checks is better. Because `(.Capability.APIVersions.Has "monitoring.coreos.com/v1/ServiceMonitor")` cannot be locally rendered. What do you think? @zwangsheng

Reply:
I'm fine with default disabled create Prometheus Operator related CRDs and remove api check.

Sounds reasonable, the user should enable this feature in the case that  Prometheus Operator related CRDs already exists, and if it does not match, throws an obvious error.

WDYT? @pan3793